File: mybookstore/lib/auth/models/store.dart
class Store {
  String? banner;
  final int id;
  String name;
  String slogan;

  Store({
    required this.banner,
    required this.id,
    required this.name,
    required this.slogan,
  });

  Store clone() {
    return Store(banner: banner, id: id, name: name, slogan: slogan);
  }

  factory Store.fromMap(Map<String, dynamic> map) {
    return Store(
      banner: map['banner'],
      id: map['id'],
      name: map['name'],
      slogan: map['slogan'],
    );
  }

  Map<String, dynamic> toMap() {
    return {'banner': banner, 'id': id, 'name': name, 'slogan': slogan};
  }
}

File: api/src/main/java/nathan/mg/api/store/Store.java
package nathan.mg.api.store;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Store {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, updatable = false)
    private LocalDateTime creationDateTime;
    
    
    private String name;
    private String slogan;
    @Column(columnDefinition = "LONGTEXT", nullable = true)
    private String banner;

    public Store(StoreDto store) {
    	this.creationDateTime = LocalDateTime.now();
        this.name = store.name();
        this.slogan = store.slogan();
        this.banner = store.banner();
    }
}
